elftools/dwarf/descriptions.py

```diff
@@ -434,6 +434,8 @@ def _data_member_location_extra(attr, die, section_offset):
     if attr.form in ('DW_FORM_data1', 'DW_FORM_data2',
                      'DW_FORM_data4', 'DW_FORM_data8'):
         return ''  # No extra description needed
+    elif attr.form in ('DW_FORM_sdata'):
```

Comment:
I did that in case there were other attributes that you were aware of that would also need to be included in this case. But if not, that’s fine, I can change it to the simpler form.

Thanks,

Brad
408-742-4764

From: Eli Bendersky [mailto:notifications@github.com]
Sent: Friday, February 26, 2016 5:28 AM
To: eliben/pyelftools <pyelftools@noreply.github.com>
Cc: Seifers, Brad (US) <brad.seifers@lmco.com>
Subject: EXTERNAL: Re: [pyelftools] Fix _data_member_location_extra to not crash (#98)


In elftools/dwarf/descriptions.py<https://github.com/eliben/pyelftools/pull/98#discussion_r54242055>:

> @@ -434,6 +434,8 @@ def _data_member_location_extra(attr, die, section_offset):

>      if attr.form in ('DW_FORM_data1', 'DW_FORM_data2',

>                       'DW_FORM_data4', 'DW_FORM_data8'):

>          return ''  # No extra description needed

> +    elif attr.form in ('DW_FORM_sdata'):

no need for "in" here, just == 'DW_FORM_sdata'

—
Reply to this email directly or view it on GitHub<https://github.com/eliben/pyelftools/pull/98/files#r54242055>.